### PR TITLE
Add shouldRender and allow a function to be passed to ignoredTags.

### LINKED ISSUE #691: Suggestion: add a property called shouldRender to renderMathInElement options
I would like to suggest adding an optional property called `shouldRender` to the `renderMathInElement` options which if provided would accept the current HTML element being considered and return a Boolean indicating whether or not to render math inside it. This would be used in place of `ignoreTags`. Alternatively you could make `ignoreTags` optionally a function in the same way but the name makes less sense. This would neatly cover any scenario for opting out of math rendering and users could come up with their own conventions (like putting the attribute `data-render-math="false"` on elements they don't want rendered).

### PR DESCRIPTION
**Pull request by leungbk**
Closes #691.

### CODE REVIEW

**Comment by leungbk** [comment]
[clabot:check]

File: contrib/auto-render/auto-render.js [hunk 1]
```diff
@@ -102,9 +120,17 @@ const renderMathInElement = function(elem, options) {
         // treating it as if it were the start of a KaTeX math zone.
         {left: "\\[", right: "\\]", display: true},
     ];
-    optionsCopy.ignoredTags = optionsCopy.ignoredTags || [
-        "script", "noscript", "style", "textarea", "pre", "code",
-    ];
+
+    if (optionsCopy.ignoredTags && optionsCopy.shouldRender) {
+        throw new Error(
+            "Please specify either ignoredTags or shouldRender, not both"
```
Comment (ylemkimon) [inline contrib/auto-render/auto-render.js]: Why not both?

Comment (leungbk) [reply contrib/auto-render/auto-render.js]: I felt it was more intuitive to specify only `.ignoredTags` (`.shouldRender`) and have everything not classified as such implicitly satisfy `.shouldRender` (`.ignoredTags`).

**Comment by leungbk** [comment]
I'll try to figure out the decrease in code coverage.

**Comment by leungbk** [comment]
It seems that even if I checkout the `master` branch, running `yarn test:jest:coverage` locally still gives only about `85%` coverage. The codecov bot on GitHub seemed to report `93%` (after reporting about `85%` yesterday) after I merged earlier, and continues to report `93%`. I'm not sure why `codecov` is still reporting `85%` on my computer, and why the codecov bot reported `85%` yesterday and `93%` today.

**Comment by ylemkimon** [comment]
@leungbk Codecov result includes coverage from screenshot tests and we don't strictly enforce code coverage changes, especially on `contrib` extensions. So you don't need to worry about them.

File: contrib/auto-render/auto-render.js (hunk 1 above)
Comment (ylemkimon) [reply contrib/auto-render/auto-render.js]: But the user may want to ignore some tags and other elements by another criterion.

Comment (leungbk) [reply contrib/auto-render/auto-render.js]: Would you provide an example of what you have in mind? The way I see it, if a user specifies both `optionsCopy.ignoredTags = ["script", "noscript", "style", "textarea", "pre", "code"]` and `optionsCopy.shouldRender = ["foo", "bar"]`, I wouldn't know what to do with a `<baz>` tag, whereas providing only one of the two properties would make it clearer.

**Comment by ylemkimon** [comment]
Closing due to inactivity. Please feel free to reopen or create a new PR.

**Pull request closed without merging**